src/strings.c

```diff
@@ -2330,7 +2333,7 @@ format_typeof(
 	    if (fmt_spec == 'p')
 		return TYPE_POINTER;
 	    else if (fmt_spec == 'b' || fmt_spec == 'B')
-		return TYPE_UNSIGNEDINT;
+		return TYPE_UNSIGNEDLONGLONGINT;
```

Comment:
Do the following tests work after this change?

```
	n = vim_snprintf(buf, bsize, "%1$0.*2$llb", (uvarnumber_T)12, 6);
	assert(n == 6);
	assert(bsize == 0 || STRNCMP(buf, "001100", bsize_int) == 0);
	assert(bsize == 0 || buf[MIN(n, bsize_int)] == '\0');

	n = vim_snprintf(buf, bsize, "%1$0.*2$b", 12, 6);
	assert(n == 6);
	assert(bsize == 0 || STRNCMP(buf, "001100", bsize_int) == 0);
	assert(bsize == 0 || buf[MIN(n, bsize_int)] == '\0');
```

Reply:
Rethinking. 'b' is a Vim-specific specifier, so it might be not mandatory to support the latter case.

Reply:
> Rethinking. 'b' is a Vim-specific specifier, so it might be not mandatory to support the latter case.

Yes, and 'b' has always been 64 bits as well. If you look at the way the format specifier is put into the source code, as a global ``char *`` and not in the sprintf() function proper, you can see that this has been done to work around a compiler warning (if you put the format string in inside the call, the compiler will warn you about the format specifier not being ll, and the argument is. Oh, no, that's not it: It complains about the 'b' specifier being unknown).

In any case, I think that the 'b' specifier is always 64 bit, because it has always been that way in the past.

I'll re-work the the 'usetvs' you mentioned, like  (what was I thinking...)